eth/hooks/engine_v2_hooks.go

```diff
@@ -28,8 +29,25 @@ func AttachConsensusV2Hooks(adaptor *XDPoS.XDPoS, bc *core.BlockChain, chainConf
 		listBlockHash[0] = currentHash
 		parentNumber := number.Uint64() - 1
 		parentHash := currentHash
+
+		// check and wait the latest block is already in the disk
+		// sometimes blocks are yet inserted into block
+		for timeout := 0; ; timeout++ {
+			parentHeader := chain.GetHeaderByNumber(parentNumber)
+			if parentHeader != nil { // found the latest block in the disk
+				break
+			}
+			log.Info("[V2 Hook Penalty] parentHeader is nil, wait block to be writen in disk", "parentNumber", parentNumber)
+			time.Sleep(200 * time.Millisecond) // 0.2s
+
+			if timeout > 50 { // wait over 10s
+				log.Error("[V2 Hook Penalty] parentHeader is nil, wait too long not writen in to disk", "parentNumber", parentNumber)
+				return []common.Address{}, fmt.Errorf("parentHeader is nil")
+			}
+		}
+
 		for i := uint64(1); ; i++ {
-			parentHeader := chain.GetHeader(parentHash, parentNumber)
+			parentHeader := chain.GetHeaderByNumber(parentNumber)
```

Comment:
why don't use parentHash?

Reply:
Because less parameter, I will check if with hash is faster then I change back

Reply:
with parentHash, it is more rigorous. without parentHash, it may be that there is fork and it get the wrong block

Reply:
hmm, good point!